src/3d/framegraph/qgsdepthrenderview.cpp

```diff
@@ -0,0 +1,130 @@
+#include <Qt3DRender/QTexture>
+#include <Qt3DRender/QClearBuffers>
+#include <Qt3DRender/qsubtreeenabler.h>
+#include <Qt3DRender/QCameraSelector>
+#include <Qt3DRender/QRenderStateSet>
+#include <Qt3DRender/QDepthTest>
+#include <Qt3DRender/QCullFace>
+#include <Qt3DRender/QRenderCapture>
+#include "qgsdepthentity.h"
+
+
+QgsDepthRenderView::QgsDepthRenderView( const QString &viewName,                     //
+                                        Qt3DRender::QCamera *mainCamera,             //
+                                        QSize size,                                  //
+                                        Qt3DRender::QTexture2D *forwardDepthTexture, //
+                                        Qt3DCore::QEntity *rootSceneEntity )
+  : QgsAbstractRenderView( viewName )
+  , mMainCamera( mainCamera )
+{
+  mLayer = new Qt3DRender::QLayer;
+  mLayer->setRecursive( true );
+  mLayer->setObjectName( mViewName + "::Layer" );
+
+  // depth rendering pass
+  buildRenderPass( size, forwardDepthTexture, rootSceneEntity );
+}
+
+void QgsDepthRenderView::updateWindowResize( int width, int height )
+{
+  mColorTexture->setSize( width, height );
+  mDepthTexture->setSize( width, height );
+}
+
+Qt3DRender::QRenderTarget *QgsDepthRenderView::buildTextures( QSize size )
+{
+  // depth buffer render to copy pass
+  // Create a render target output for rendering color.
+  Qt3DRender::QRenderTargetOutput *colorOutput = new Qt3DRender::QRenderTargetOutput;
+  colorOutput->setAttachmentPoint( Qt3DRender::QRenderTargetOutput::Color0 );
+
+  // Create a texture to render into.
+  mColorTexture = new Qt3DRender::QTexture2D( colorOutput );
+  mColorTexture->setSize( size.width(), size.height() );
+  mColorTexture->setFormat( Qt3DRender::QAbstractTexture::RGB8_UNorm );
+  mColorTexture->setMinificationFilter( Qt3DRender::QAbstractTexture::Linear );
+  mColorTexture->setMagnificationFilter( Qt3DRender::QAbstractTexture::Linear );
+  mColorTexture->setObjectName( mViewName + "::mColorTexture" );
+
+  // Hook the texture up to our output, and the output up to this object.
+  colorOutput->setTexture( mColorTexture );
+
+  Qt3DRender::QRenderTargetOutput *depthOutput = new Qt3DRender::QRenderTargetOutput;
+
+  depthOutput->setAttachmentPoint( Qt3DRender::QRenderTargetOutput::Depth );
+  mDepthTexture = new Qt3DRender::QTexture2D( depthOutput );
+  mDepthTexture->setSize( size.width(), size.height() );
+  mDepthTexture->setFormat( Qt3DRender::QAbstractTexture::DepthFormat );
+  mDepthTexture->setMinificationFilter( Qt3DRender::QAbstractTexture::Linear );
+  mDepthTexture->setMagnificationFilter( Qt3DRender::QAbstractTexture::Linear );
+  mDepthTexture->setComparisonFunction( Qt3DRender::QAbstractTexture::CompareLessEqual );
+  mDepthTexture->setComparisonMode( Qt3DRender::QAbstractTexture::CompareRefToTexture );
+  mDepthTexture->setObjectName( mViewName + "::mDepthTexture" );
+
+  depthOutput->setTexture( mDepthTexture );
```

Comment:
Actually the depth texture should not be needed here at all and could be removed - we only need to draw to the color texture...

Reply:
OK but the goal of this PR was to just extract the depth renderview.

I will do an extra cleanup as you suggest.